nsidc-data-cookbook/how-to-guides/get_latitude_and_longitude.qmd

```diff
@@ -0,0 +1,47 @@
+---
+title: "How to get latitude and longitude for the grid cells of a NetCDF file"
+author: Andrew P. Barrett
+date: last_modified
+---
+
+## Problem
+My data are in a NetCDF file but there are no latitude and longitude coordinates.  How do I get the latitude and longitude for each grid cell center?
+
+## Solution
+```
+import numpy as np
+
+import rioxarray
+import xarray as xr
+from pyproj import CRS, Transform
+
+
+# Load dataset using decode_coords='all'
+ds = xr.open_dataset('example_data/NSIDC0081_SEAICE_PS_N25km_20230627_v2.0.nc',
+                     decode_coords='all')
+
+# Instantiate source pyproj CRS from ds.rio.crs object
+source_crs = CRS(ds.rio.crs.to_wkt())
+# Instantiate destination pyproj CRS using EPSG code for WGS84 
+destination_crs = CRS(4326)
+
+# Instantiate pyproj transformer instance using source and destination CRS
+transform = Transformer.from_crs(source_crs, destination_crs)
+
+# Create 2D arrays of x and y coordinates
+x2d, y2d = np.meshgrid(ds.x, ds.y)
+
+# Calculate latitude and longitudes for each grid cell
+lat, lon = transformer.transform(x2d, y2d)
```

Comment:
```suggestion
lat, lon = transform.transform(x2d, y2d)
```

Reply:
The script ran for me using an NSIDC-0081 example file after I made these changes.